### PR TITLE
Update documentation about comments to include mention of hashbangs.

### PR DESCRIPTION
**Pull request by kotnen**
### Description

The documentation previously lacked a mention of how hashbangs are handled.  Now, the documentation mentions that hashbangs are excepted from the comment spacing rules.  This fixes issue #3786.

### Checklist - did you ...

- [x] Add an entry in `CHANGES.md` if necessary?
- [x] Add / update tests if necessary?
- [x] Add new / update outdated documentation?

### CODE REVIEW

File: docs/the_black_code_style/current_style.md [hunk 1]
```diff
@@ -254,11 +254,12 @@ required due to an inner function starting immediately after.
 
 _Black_ does not format comment contents, but it enforces two spaces between code and a
 comment on the same line, and a space before the comment text begins. Some types of
-comments that require specific spacing rules are respected: doc comments (`#: comment`),
-section comments with long runs of hashes, and Spyder cells. Non-breaking spaces after
-hashes are also preserved. Comments may sometimes be moved because of formatting
-changes, which can break tools that assign special meaning to them. See
-[AST before and after formatting](#ast-before-and-after-formatting) for more discussion.
+comments that require specific spacing rules are respected: hashbangs (`#! comment`),
```
Comment (hauntsaninja) [inline docs/the_black_code_style/current_style.md:257]:
```suggestion
comments that require specific spacing rules are respected: shebangs (`#! comment`),
```

**Review by hauntsaninja — APPROVED** [review]
Thanks! Suggesting using "shebangs" since that's the more common term (as per wikipedia)

File: docs/the_black_code_style/current_style.md (hunk 1 above)
Comment (kotnen) [reply docs/the_black_code_style/current_style.md]: Thanks for the suggestion :)

**Comment by hauntsaninja** [comment]
Thank you!

**Pull request merged**